src/runtime_safe_heap.js

```diff
@@ -106,11 +106,12 @@ function SAFE_HEAP_STORE(dest, value, bytes, isFloat) {
 #if SAFE_HEAP_LOG
   out('SAFE_HEAP store: ' + [dest, value, bytes, isFloat, SAFE_HEAP_COUNTER++]);
 #endif
+  var brk = _sbrk() >>> 0;
   if (dest <= 0) abort('segmentation fault storing ' + bytes + ' bytes to address ' + dest);
   if (dest % bytes !== 0) abort('alignment error storing to address ' + dest + ', which was expected to be aligned to a multiple of ' + bytes);
-  if (dest + bytes > HEAPU32[DYNAMICTOP_PTR>>2]) abort('segmentation fault, exceeded the top of the available dynamic heap when storing ' + bytes + ' bytes to address ' + dest + '. DYNAMICTOP=' + HEAP32[DYNAMICTOP_PTR>>2]);
-  assert(DYNAMICTOP_PTR);
-  assert(HEAP32[DYNAMICTOP_PTR>>2] <= HEAP8.length);
+  if (dest + bytes > brk) abort('segmentation fault, exceeded the top of the available dynamic heap when storing ' + bytes + ' bytes to address ' + dest + '. DYNAMICTOP=' + brk);
+  assert(brk);
```

Comment:
This assert is no long meaningful I think since we are no longer dealing with a pointer here.

Reply:
I think it still has value. 0 is never a valid break location, and may indicate that we've failed to initialize sbrk properly.

In fact this could be expanded into `assert(brk >= STACK_BASE)` since the break has to be after globals and the stack. Or is that too much?

Reply:
Adding the stronger assert now, but I can remove it if you think I'm missing something.